tests/federation_encryption_test.go

```diff
@@ -0,0 +1,63 @@
+package tests
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+)
+
+func TestEncryption(t *testing.T) {
+	emptyString := ""
+
+	deployment := Deploy(t, b.BlueprintFederationOneToOneRoom)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	bob := deployment.Client(t, "hs2", "@bob:hs2")
+
+	encAlice := client.MustCreateEncryptedClient(t, alice)
+	encBob := client.MustCreateEncryptedClient(t, bob)
+
+	// start syncing using mautrix.Client, needed to update the olmMachine store
+	go encAlice.Sync(t)
+	go encBob.Sync(t)
+	defer encAlice.StopSync()
```

Comment:
I really dislike this from an API perspective because it's so easy to forget to do this (either sync, sync in a goroutine, defer stop on _both_ clients, etc). It would be better if we encapsulated this in `MustCreateEncryptedClient` where we just return `encAlice, finish` or something and then just `defer finish()`, which is similar to the usage of `federation.Server`:
```go
	cancel := srv.Listen()
	defer cancel()
```